### PR TITLE
Migrate React Native CLI (RN v0.60.x)

### PR DESCRIPTION
**Pull request by nineto4**
Migrate React Native CLI
https://github.com/react-native-community/cli/blob/master/docs/configuration.md#migration-guide

### CODE REVIEW

**Comment by congnguyen91** [comment]
+1 this should be checked :)

**Comment by v0lume** [comment]
can it be merged asap cause i don't want to see this errors anymore?

**Comment by aarongrider** [comment]
Bump on this!

**Comment by namvoeh** [comment]
@nineto4 it's missing `postlink` and `postunlink` hooks, please check my PR https://github.com/microsoft/react-native-code-push/pull/1699

**Comment by birkir** [comment]
Cool, `react-native-code-push` is currently blocking RN 0.60 and 0.61 users from using the auto-linking feature.

@alexandergoncharov can you merge and release a new version?

File: react-native.config.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+module.exports = {
+  project: {
+    ios: {
+      sharedLibraries: [
+        "libz"
+      ]
+    },
+    android: {
```
Comment (friederbluemle) [inline react-native.config.js]: Minor question, but any reason [`android`, `ios`] are no longer in alphabetical order (like in the original file)?

Comment (friederbluemle) [reply react-native.config.js]: I understand it does not matter from a program execution point of view. It matters from a code cleanliness, maintenance, VCS point of view (the order has been changed from the original code)

Comment (namvoeh) [reply react-native.config.js]: ah I see

**Comment by namvoeh** [comment]
Please help to merge this PR I took around 2 months to got merged

**Comment by giacomocerquone** [comment]
@builtbyproxy actually I believe that the use of prelink and postunlink "reserved" words in the commands object relies on a quite old rnpm version (in fact you can still find it in their locked repo https://github.com/rnpm/rnpm).
So it should be something that was already moved to the hooks object.

But I want to make clear that these are all assumptions and I couldn't find absolutely anything about how rnpm should be configured for a native lib. The only thing that drives me to that conclusion is the rnpm doc (showing an old commands-prelink/postlink configuration) and the migration guide (showing a prelink field inside the hooks object https://github.com/react-native-community/cli/blob/master/docs/configuration.md#libraries)

**Comment by builtbyproxy** [comment]
@giacomocerquone Absolutely Fantastic! Thanks a lot for explaining all of that!
Do you know what's blocking this merge?

**Comment by giacomocerquone** [comment]
@builtbyproxy the stale situation of this repo is blocking the merge.
I read comments of almost a year ago that the project went under a heavy modification process by the core team and I kept reading of people who's claiming that the project is not dead... yeah, probably, but surely I don't think this is the proper way to handle a migration. No public WIP board, no way we can help since the work is not being done through this repo... this doesn't look like OSS honestly 😅

To make people happy, it was as simple as declaring 1 or 2 trustable people that would have taken care of updating the docs, merging pull request of few lines like this etc. Etc.

Don't want to disrespect with this comment... we all rely on this package, and the devs did and I'm sure are still doing a great work, but just wanted to note this in order to handle the next "project rework" there might be a little bit better.

Honestly I'd like to help managing the issues and the pull requests, but no one would give me credit nor permission 😊

**Comment by bitlab-code** [comment]
Absolutely agree with what @giacomocerquone says
Without wishing to be pessimistic, everything suggests that Microsoft no longer has an interest in CodePush and React Native. Too much silence is probably "The beginning of the end", perhaps this "next version of CodePush" will be a new appcenter service

**Comment by giacomocerquone** [comment]
Well @bitlab-code I wouldn't say that. React Native seems to be very important to Microsoft*, I just said that things could have done in a cleaner and more proper way.

*https://www.theregister.co.uk/2019/11/07/microsoft_react_native/
*https://twitter.com/reactiflux/status/1184927584502665217

**Comment by builtbyproxy** [comment]
@giacomocerquone Who do we need to bug to get this stuff approved? 
My patch-package for this is starting to get a bit overwhelming

**Comment by giacomocerquone** [comment]
Let's just ping the main contributors: @geof90 @lostintangent @max-mironov @alexandergoncharov the community needs you, I hope you won't be bothered by this request!
Thank you very much for consideration.

**Comment by joseph-d-p** [comment]
See PR #1735

**Comment by builtbyproxy** [comment]
Opened by a member of Microsoft :O

**Comment by alexandergoncharov** [comment]
Hi all,
Thanks for contributing and sorry for this big delay!

I think that it was fixed a long time ago by https://github.com/microsoft/react-native-code-push/pull/1735.
So, I'm going to close this PR for now. Please feel free to reopen it if you have any questions!

**Pull request closed without merging**